### DISCUSSION TITLE
MOSE ms-104cz - specific parameter type of external switch

### DISCUSSION THREAD

**Discussion by bodbod44** [Koenkk/zigbee2mqtt#24811] [Devices]
Hello,

I have a MOES MS-104CZ zigbee module (TS0003 / _TZ3000_pfc7i3kt).

It is well recognized by zigbee2mqtt.

However, it works with a rocker switch (2 positions / two-way) and **I want to make it work with a push button**.

In the manual, it is compatible
I sent an email to MOES who tells me that it is configured in the mobile app (type of external switch).

But, I do not find this option in zigbee2mqtt on the device.

I do not have a Tuya gateway

Is it possible to add the option and how?

Thanks

**Comment by bodbod44** [comment]
Hello. What tool does the screenshot come from? I am on jeedom/zigbee2mqttLe 18 déc. 2024 17:13, bastoonch ***@***.***> a écrit :
@bodbod44
were you able to do it without the app ? I am looking at doing the same thing.
I am wondering if we can change that setting
image.png (view on web)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
![397007157-006c7acb-7e23-4304-a7f8-0abd0aea337e](https://github.com/user-attachments/assets/70f5b83f-fda1-46e5-bedb-5a77547b9f69)

**Reply by bastoonch** [reply]
my bad. I had several tabs open at the same time and posted on the wrong one.
FYI, it is on Home Assistant and probably you should have the same possibility to configure the device itself in Jeedom.

https://community.home-assistant.io/t/momentary-switch-configuration-with-zha/410937/9

**Reply by bodbod44** [reply]
on jeedom/zigbee2mqtt, I have this
![Capture1](https://github.com/user-attachments/assets/c3b2bad1-4440-4bf7-a904-8f5d43eaf28f)

![Capture2](https://github.com/user-attachments/assets/b3f575e8-3aed-4d25-b569-da251f75bea8)

![Capture3](https://github.com/user-attachments/assets/a31285a3-2151-48e2-8085-793f865f5d92)

![Capture4](https://github.com/user-attachments/assets/b586ff97-ec64-4167-a961-08ffbdf5ee52)

**Reply by bastoonch** [reply]
I tried on my z2m dev instance and I am facing the same behavior. It seems the device isn't fully implemented on Z2M. I found interesting article about external converter...maybe the way to go.
https://github.com/Koenkk/zigbee2mqtt/issues/9465

**Reply by bastoonch** [reply]
it worked for me. 

![image](https://github.com/user-attachments/assets/6ec11660-ac13-47f5-a2a5-a0e8cb9b2994)

![image](https://github.com/user-attachments/assets/ddacd2df-d354-41ab-a467-045b6c2b28a7)

![image](https://github.com/user-attachments/assets/5cf9c4d5-a871-4f90-8795-8a47b19a862a)

restarted Z2M and read the value again and it was fine.

![image](https://github.com/user-attachments/assets/cbf7554b-90c0-4f87-99c3-7414842f1fef)

**Reply by bodbod44** [reply]
I can change the setting.

However, I can't with endpoint 2 and 3

![Capture d'écran 2024-12-20 084449](https://github.com/user-attachments/assets/323d25ed-379e-402b-b876-be082e2d512a)

**Reply by bastoonch** [reply]
It has to be done only for endpoint 1. 
It changed the global behavior of the switch for me.

**Reply by bodbod44** [reply]
OK, I will test

**Reply by bodbod44** [reply]
it work

**Reply by bastoonch** [reply]
Great to hear.
